### ISSUE TITLE
Release 2.2

### ISSUE THREAD

**Issue by bradjc** [tock/tock#3197]
This issue serves as a tracker for remaining tasks to tag a 2.2 release candidate, and as a tracking mechanism for release testing once that is done. 

## Blocking Tasks
- [x] Merge remaining timer PRs (@lschuermann )
- [ ] Update changelog (@hudson-ayers )
- [ ] Update crate version numbers (see https://github.com/tock/tock/issues/3197#issuecomment-2248183406)
- [ ] Test

## Notable New Features in 2.2
- AppID (#2809)
- CortexM HardFaultHandler bug (#3798)
- StreamingProcessSlice (#4208)
- TODO -- fill out

## Test boards:

- [ ] acd52832
- [ ] apollo3
- [ ] arty_e21
- [ ] clue_nrf52840
- [ ] esp32-c3
- [ ] hail
- [ ] hifive1
- [ ] hifive_inventor
- [ ] imix
- [ ] imxrt1050-evkb
- [ ] litex
  - [ ] sim
  - [ ] fpga
- [ ] makepython-nrf528340
- [ ] microbit_v2
- [ ] msp_exp432p401r
- [ ] nano33ble
- [ ] nano33ble_rev2
- [ ] nano_rp2040_connect
- [ ] nordic
  - [ ] nrf52dk
  - [ ] nrf52840dk
  - [ ] nrf52840_dongle
- [ ] nucleo_f429zi
- [ ] nucleo_f446re
- [ ] opentitan
  - [ ] earlgrey-cw310
- [ ] particle_boron
- [ ] pico_explorer_base
- [ ] qemu_rv32_virt
- [ ] raspberry_pi_pico
- [ ] redboard_redv
- [ ] sma_q3
- [ ] stm32f3discovery
- [ ] stm32f412gdiscovery
- [ ] stm32f429idiscovery
- [ ] swervolf
- [ ] teensy40
- [ ] weact_f401ccu6
- [ ] wm1110dev
- [ ] tutorials
  - [ ] nrf52840dk-thread-tutorial

**Comment by bradjc** [comment]
Also, I think that now is the time to have the discussion about deprecating boards. That way we have time to 1) make sure they should actually be deprecated, and 2) move them into tock-archive properly.

**Comment by lschuermann** [comment]
I would like to see #2977 and the proposed DMABuffer design -- which would also incorporate the mutable/immutable buffer handling -- to be on the roadmap for the next release. I realize those two are on my table, I'll work on getting them ready, just to have that be on the radar.

**Comment by hudson-ayers** [comment]
> Also, I think that now is the time to have the discussion about deprecating boards. That way we have time to 1) make sure they should actually be deprecated, and 2) move them into tock-archive properly.

The following boards did not receive release testing:
- imxrt1050-evkb 
- msp_exp432p401r
- arty_e21
- acd52832
- clue_nrf52840

The one other board we might want to consider deprecating is the weact board, since the version we support has been discontinued, though we do have someone available to perform testing for it if we decide to keep it.

**Comment by alexandruradovici** [comment]
I tested the clue_nrf52840 and it passed all tests. It is one of the boards that is fully supported, it works perfectly with tockloader and tock bootloader.

I wanted to suggest placing it in Tier 1.

imxrt1050-evkb was tested by @valexandru.

**Comment by hudson-ayers** [comment]
Ah sorry I must have missed the testing list for those two boards

**Comment by brghena** [comment]
https://github.com/tock/tock/pull/3798 feels important to me. And it'll be merged fairly soon

**Comment by hudson-ayers** [comment]
After the call today we had some brief discussion about where this release should land relative to the updated Yield WaitFor PR.

I would like to propose that we reduce the required amount of release testing for minor releases. We should only block on running the full suite of release tests for Imix + nrf52840dk, plus any automated tests that we can run through Leon's hardware CI system, and then set a one week timer for additional testing. Perhaps this would require some updates to our tiered board system. 

At a high level, I think our current approach discourages us too much from making new releases, which has various downsides for the project. 

We would still require a full suite of release testing for major version increases. Happy to hear what others think about this approach!

**Comment by bradjc** [comment]
When we do a release, why do we not update the version numbers in the crates? Is there a strategic reason to not do that, or is it more because we have so many crates and there isn't a particularly good reason to go through the effort?

**Comment by hudson-ayers** [comment]
I thought it was because updating the version numbers of the crates above 1.0 could imply certain semver guarantees for those crates that we are not ready to make. For example, we aren't ready to commit to no breaking changes of any of the public APIs in the arch crate

**Comment by bradjc** [comment]
> version numbers of the crates above 1.0 could imply certain semver guarantees

Oh right, that makes sense.

I want to suggest we consider including the version number with a prepended 0 (i.e., this release would be `0.2.2` in the crates). That's a little odd, but so is always using the version `0.1.0`.

**Comment by lschuermann** [comment]
Superseded by #4272.
